integr8ly/cloud-resource-operator
# MGDAPI-2160 - apply critical security updates to Elasticache

## Overview
Jira: https://issues.redhat.com/browse/MGDAPI-2160

First commit contains a refactor of the cro_redis_service_maintenance metric. It will now provide information about service updates related to a specific Redis instance. This also allows us to drop one permission and work around this issue - https://issues.redhat.com/browse/CCO-107

Second commit implements new logic for applying critical security updates.
These updates will only be applied if `.spec.applyImmediately` is set to `true`.

## Verification
- Build the operator image, bundle and index
- Deploy to an AWS cluster using RHOAM (clusterStorage = false)
- Verify that RHOAM installation finishes successfully
- Verify that CRO doesn't crash
- Verify that cro_redis_service_maintenance metric produces relevant information
- Add this element into CRO configuration/strategy ConfigMap, into .redis.production - `"serviceUpdates": ["elasticache-20210615-002"]` and verify that CRO doesn't crash
- On all Redis CRs set `.spec.applyImmediately = true` and verify again that CRO doesn't crash
- Review logs - you should see some new informational messages about service updates

## Checklist
